### PR TITLE
Improved single file monitoring.

### PR DESCRIPTION
**Pull request by Mr-indescribable**
Hi,

I found that the `Inotify` class is not doing a great job in single file monitoring.

According to `man inotify` : 

```
           IN_IGNORED
                  Watch  was removed explicitly (inotify_rm_watch(2)) or auto‐
                  matically (file was deleted, or filesystem  was  unmounted).
```

The event `IN_IGNORED` is generated when a watch has been automatically removed, this happens when the monitored file is removed, and it also happens when user modify the monitored file with text editors that will replace the original file while saving (such as `vim`). 

But the `Inotify` class itself doesn't handle this event, so the consequence is: when this happens, the watch has been automatically removed by the kernel but it still exists in the `Inotify` class (in `Inotify.__watches` and `Inotify.__watches_r`), and this causes a bug, when a monitored file is removed/replaced, users can no long re-add it back into the monitoring, if they try to remove it first by calling `Inotify.remove_watch()` then the system call `inotify_rm_watch()` will return an error because the watch does not exist in the kernel space.

So, the thing I've done is giving users an optional argument to make the `Inotify` class handle the `IN_IGNORED` event and it's disabled by default.

And I changed the usage of `superficial` parameter in `Inotify.remove_watch` as well which was not being used.

### CODE REVIEW

File: inotify/adapters.py [hunk 1]
```diff
@@ -231,6 +231,10 @@ def event_gen(
                         in self._handle_inotify_event(fd):
                     last_hit_s = time.time()
 
+                    if handle_in_ignored and\
```
Comment (dsoprea) [inline inotify/adapters.py:234]: Just make it default behavior. The user shouldn't have a choice. Reword the commit message to mention that we're specifically handling the IN_IGNORED event, now.

File: inotify/adapters.py [hunk 2]
```diff
@@ -115,7 +115,7 @@ def remove_watch(self, path, superficial=False):
 
         del self.__watches[path]
 
-        self.remove_watch_with_id(wd)
+        self.remove_watch_with_id(wd, superficial)
```
Comment (dsoprea) [inline inotify/adapters.py:118]: Put this in a separate commit since it's not related to this commit.

Comment (dsoprea) [reply inotify/adapters.py:118]: Already resolved by #57

**Comment by Mr-indescribable** [comment]
Hi, I reverted the original commit and did what you asked, but beware, this changes the default behavior of `Inotify` class and could bring some backward compatibility issue.

**Comment by Mr-indescribable** [comment]
Hi, I reverted the original commit and did what you asked, but beware, this changes the default behavior of `Inotify` class and could bring some backward compatibility issue.
